Add ReadOnlySpan<byte> overloads

chwarr opened the pull request:
Adds a ReadOnlySpan<byte> overload. The Span<byte> overload forwards to
the ReadOnlySpan<byte> implementation.

Adding a new overload maintains binary and reflection compability,
whereas just changing the parameter type is only source compatible.

chwarr committed:
Add ReadOnlySpan<byte> overloads

Adds a ReadOnlySpan<byte> overload. The Span<byte> overload forwards to
the ReadOnlySpan<byte> implementation.

Adding a new overload maintains binary and reflection compability,
whereas just changing the parameter type is only source compatible.

chwarr commented on src/MurmurHash.Net/MurmurHash3.cs:
```diff
@@ -5,7 +5,13 @@ namespace MurmurHash.Net
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static uint Hash32(Span<byte> bytes, uint seed)
+        {
```
If you don't care about binary backward comparability, this can be a much simpler change. The type of `bytes` just needs to be changed from `Span<byte>` to `ReadOnlySpan<byte>`. Let me know if you're OK with the simpler version and I can revise.

odinmillion replied:
Great! Let's change `Span` to `ReadOnlySpan`. It is not necessary to be able to modify data inside the method.

chwarr replied:
Revised to just be the single change from `Span` to `ReadOnlySpan`.

chwarr committed:
Simplify ReadOnlySpan change: only one method

Since we don't care about binary compatibility and changing the method
argument from Span to ReadOnlySpan is compatible at compile time, we
just need one method instead of an overload set.

Using https://sharplab.io/ I compared the generated x64 release assembly
for the Span and ReadOnlySpan implementations: they are identical.
